### DISCUSSION TITLE
Signature of DrawTriangles() - what is img for?

### DISCUSSION THREAD

**Discussion by tomekc** [hajimehoshi/ebiten#1434] [Q&A] [answered]
Hello, 

I have trouble understanding the method signature. In my use case, I just want to draw vector lines in the specified color.
What is the purpose of `img` argument? It is not stated clearly in the comments.

I ended up giving a random image for this argument because even though it is a pointer, the program crashes when it is null.

```
func (i *Image) DrawTriangles(vertices []Vertex, indices []uint16, img *Image, options *DrawTrianglesOptions)
```

**Comment by hajimehoshi** [comment, marked as answer]
Hi,

The given image is used in the same manner as [`DrawImage`](https://pkg.go.dev/github.com/hajimehoshi/ebiten/v2#Image.DrawImage). The image is rendered on the triangles specified by the given vectors and indices. The texture coordinates are specified by the vertices. See [`Vertex` definition](https://pkg.go.dev/github.com/hajimehoshi/ebiten/v2#Vertex).

**Reply by tomekc** [reply]
Thanks. So this is intended to draw a textured polygon specified by triangles?

In that case, is there a way to draw primitive shapes, like lines, maybe rectangles?

**Reply by hajimehoshi** [reply]
Good question. I'd use a white empty image. See https://github.com/hajimehoshi/ebiten/blob/master/examples/polygons/main.go

**Reply by tomekc** [reply]
Thanks, a bit hacky, but still a solution :) 

In the example, I see that you created an empty 3x3 image, and during the drawing, a routine used sub-image `1,1 - 2,2`. I guess that was necessary?

**Reply by hajimehoshi** [reply]
> Thanks, a bit hacky, but still a solution :)

I admit this seems hacky but I chose the ease of expecting the API behavior.

Another suggestion is to use `DrawTrianglesWithShader` with your custom shader, but I think this is overkill.

> In the example, I see that you created an empty 3x3 image, and during the drawing, a routine used sub-image 1,1 - 2,2. I guess that was necessary?

Yes. Otherwise, the edges would be missing.
